src/Lean/Elab/Tactic/BuiltinTactic.lean

```diff
@@ -83,7 +86,7 @@ def addCheckpoints (stx : Syntax) : TacticM Syntax := do
           mkNode ``tacticSeq #[
             mkNode ``tacticSeq1Indented #[
               -- HACK: null node is not a valid tactic, but prevents infinite loop
-              mkNullNode (currentCheckpointBlock.push tac)
+              mkNullNode (currentCheckpointBlock.push (mkNullNode #[tac]))
```

Comment:
The outer `mkNullNode` is actually necessary, the hack was not necessary before because it was masked by the off-by-one bug